internal/host/validator.go

```diff
@@ -1626,3 +1628,21 @@ func (v *validator) noSkipMissingDisk(c *validationContext) (ValidationStatus, s
 	}
 	return ValidationSuccess, successMessage
 }
+
+func (v *validator) serviceCanConnectToSpokeKubeAPI(c *validationContext) (ValidationStatus, string) {
+	// This only applies to scenarios where we have a cluster, kube-api is enabled and the cluster is in "Day2" mode.
+	if c.cluster == nil || !c.kubeApiEnabled || *c.cluster.Status != models.ClusterStatusAddingHosts {
+		return ValidationSuccessSuppressOutput, "Host is not part of a Day2 cluster, no need to check KubeAPI connectivity"
+	}
+
+	// Validate that the service can connect to kube-api - this will be needed for some Day2 operations.
+	// This will only verify host/port connectivity.
+	apiAddress := common.GetAPIHostname(c.cluster)
+	apiHostPort := net.JoinHostPort(apiAddress, "6443")
+	_, err := net.DialTimeout("tcp", apiHostPort, 10*time.Second)
```

Comment:
What is the reason we do a simple TCP connection instead of actually checking we have full, privileged API access?

Reply:
I guess we should, I tried to keep things quick and simple here but for accuracy, we should make sure we can make an API call. I will look into this.